src/CommunityToolkit.Aspire.Hosting.SurrealDb/SurrealDbBuilderExtensions.cs

```
@@ -0,0 +1,259 @@
+    /// <code lang="csharp">
+    /// var builder = DistributedApplication.CreateBuilder(args);
+    ///
+    /// var db = builder.AddSurrealServer("surreal")
+    ///   .AddNamespace("ns")
+    ///   .AddDatabase("db");
+    /// var api = builder.AddProject&lt;Projects.Api&gt;("api")
+    ///   .WithReference(db);
+    ///  
+    /// builder.Build().Run(); 
+    /// </code>
+    /// </example>
+    /// </remarks>
+    public static IResourceBuilder<SurrealDbDatabaseResource> AddDatabase(
+        this IResourceBuilder<SurrealDbNamespaceResource> builder, 
+        [ResourceName] string name, 
+        string? databaseName = null
+    )
+    {
+        ArgumentNullException.ThrowIfNull(builder);
+        ArgumentException.ThrowIfNullOrEmpty(name);
+
+        // Use the resource name as the database name if it's not provided
+        databaseName ??= name;
+
+        builder.Resource.AddDatabase(name, databaseName);
+        var surrealServerDatabase = new SurrealDbDatabaseResource(name, databaseName, builder.Resource);
+
+        SurrealDbClient? surrealDbClient = null;
+
+        builder.ApplicationBuilder.Eventing.Subscribe<ConnectionStringAvailableEvent>(surrealServerDatabase, async (@event, ct) =>
+        {
+            var connectionString = await surrealServerDatabase.ConnectionStringExpression.GetValueAsync(ct).ConfigureAwait(false);
+            if (connectionString == null)
+            {
+                throw new DistributedApplicationException($"ConnectionStringAvailableEvent was published for the '{surrealServerDatabase}' resource but the connection string was null.");
+            }
+            
+            surrealDbClient = new SurrealDbClient(connectionString);
+        });
+
+        string namespaceName = builder.Resource.Name;
+        string serverName = builder.Resource.Parent.Name;
+        
+        string healthCheckKey = $"{serverName}_{namespaceName}_{name}_check";
+        builder.ApplicationBuilder.Services.AddHealthChecks()
```

Would it be cleaner to use `AddTypeActivatedCheck` for registering the `SurrealDbHealthCheck` and pass in the `surrealServerDatabase` as the argument for the constructor, then have the health check resolve the client?

That way if the client fails to resolve you can mark it as unhealthy, rather than having it as nullable and then forcing it to be non-null on the constructor.